packages/@storefront/flux-capacitor/test/unit/core/requests/utils.ts

```diff
@@ -479,6 +479,39 @@ suite('requests helpers', ({ expect, stub, spy }) => {
     });
   });
 
+  describe('productDetails()', () => {
+    const state: any = {
+      area: 'gurjoth',
+      collection: 'charles',
+    };
+
+    const request = {
+      area: 'joey',
+      collection: 'carlo',
+    };
+
+    const history = {
+      request: {
+        area: 'joey',
+        collection: 'carlo',
+      },
+    };
+
+    const overrides = request;
+
+    beforeEach(() => {
+      stub(RequestHelpers, 'search').withArgs(state, overrides).returns(request);
+
+      stub(Selectors, 'history').withArgs(state).returns(history);
+      stub(Selectors, 'area').withArgs(state).returns(state.area);
+      stub(Selectors, 'collection').withArgs(state).returns(state.collection);
+    });
+
+    it('should create a product details request', () => {
```

Comment:
Based on the mock data above, this should read something like: 'should create a product details request using `history`'. I think you need an accompanying test for cases where the `history` data is not set/does not exist.

Reply:
Will add.